pkg/builder/webhook.go

```diff
@@ -169,24 +176,24 @@ func (blder *WebhookBuilder) registerDefaultingWebhook() {
 }
 
 func (blder *WebhookBuilder) getDefaultingWebhook() *admission.Webhook {
-	if defaulter := blder.customDefaulter; defaulter != nil {
-		w := admission.WithCustomDefaulter(blder.mgr.GetScheme(), blder.apiType, defaulter)
-		if blder.recoverPanic != nil {
-			w = w.WithRecoverPanic(*blder.recoverPanic)
-		}
-		return w
+	var w *admission.Webhook
+	if handler := blder.mutationHandler; handler != nil {
+		w = admission.WithObjectHandler(blder.mgr.GetScheme(), blder.apiType, handler)
+	} else if defaulter := blder.customDefaulter; defaulter != nil {
+		w = admission.WithCustomDefaulter(blder.mgr.GetScheme(), blder.apiType, defaulter)
+	} else if defaulter, ok := blder.apiType.(admission.Defaulter); ok {
+		w = admission.DefaultingWebhookFor(blder.mgr.GetScheme(), defaulter)
```

Comment:
I know you are not waiting on this [PR](https://github.com/kubernetes-sigs/controller-runtime/pull/2877) to merge but we are removing the Defaulter interface.  

Not that this has anything to do with the logic you present, just calling it out in case this changes how you approach this.

Reply:
It sounds like a good thing, but yeah, it doesn't affect this PR (other than the necessary rebase).

Reply:
#2877 has been merged node